multi-agent/lib/mas/session/service.py

```
@@ -382,6 +391,40 @@ def list_user_sessions(self, identity: Identity) -> list:
+        meta = doc.get("metadata", {})
+        last_active = doc.get("run_context", {}).get("last_active_at", "")
+        cost_updated = meta.get("cost_updated_at", "")
+        if "total_cost" in meta and (not last_active or cost_updated >= last_active):
+            return
+        session_id = doc.get("run_id", "")
+        if not session_id:
+            return
+        try:
+            cost = self._tracing.get_session_cost(session_id)
+            meta["total_cost"] = cost
+            meta["cost_updated_at"] = last_active or datetime.utcnow().isoformat()
```

Ruff (0.16.1)

[warning] `datetime.datetime.utcnow()` used

(DTZ003)